### PR TITLE
Add some comments to VectorTools::interpolate_boundary_values().

### PR DESCRIPTION
**Pull request by bangerth**
/rebuild

### CODE REVIEW

File: include/deal.II/numerics/vector_tools_boundary.h [hunk 1]
```diff
@@ -95,6 +96,42 @@ namespace VectorTools
...
+   *   same order as the other error terms in the finite element method,
+   *   though there are some subtle differences when measuring the error in
+   *   the $L^2$ norm. For some details, see @cite Bartels2004 .
+   *
+   * @note An alternative to using the interpolant,
+   *     @f[ u_h|_{\partial\Omega}=I_h^{\partial\Omega} g @f]
+   *   is to use the *projection* of the boundary values $g$ onto the
+   *   finite element space on the boundary:
+   *     @f[ u_h|_{\partial\Omega}=\Pi_h^{\partial\Omega} g. @f]
+   *   The projection is available using the project_boundary_values()
```
Comment (tjhei) [inline include/deal.II/numerics/vector_tools_boundary.h:128]: I think it is worth pointing out that interpolation only works for nodal FEM spaces, while projection works in the more generic situation.

File: include/deal.II/numerics/vector_tools_boundary.h [hunk 2]
```diff
@@ -410,6 +414,14 @@ namespace VectorTools
    * vector component of the finite element used in @p dof. This entry is the
    * component number in @p boundary_functions that should be used for this
    * component in @p dof. By default, no remapping is applied.
+   *
+   * @note Using the *projection* rather than the *interpolation* of boundary
+   *   values makes relatively little difference in practice. That said,
```
Comment (tjhei) [inline include/deal.II/numerics/vector_tools_boundary.h:419]: Same here, ", except that interpolation only works for nodal FEM spaces".

**Review by tjhei — COMMENTED** [review]
Makes sense. Just a minor addition, if you are willing to humor me.

**Comment by bangerth** [comment]
@tjhei Very good points! Fixed!

**Review by tjhei — APPROVED** [review]

**Pull request merged**